src/compiler/scala/reflect/quasiquotes/Parsers.scala

```diff
@@ -153,9 +152,8 @@ trait Parsers { self: Quasiquotes =>
 
       override def refineStat(): List[Tree] =
         if (isHole && !isDclIntro) {
-          val result = RefineStatPlaceholder(in.name) :: Nil
           in.nextToken()
-          result
+          return RefineStatPlaceholder(in.name) :: Nil
```

Comment:
@Kobenko what is the motivation behind this change? How can we know that `in.nextToken()` doesn't change `in.name`?

Reply:
@lrytz you are right, this change is misleading, if you don't know what `nextToken()` is a side effect